sammba/registration/struct.py

```diff
@@ -595,22 +594,38 @@ def anats_to_common(anat_filenames, write_dir, brain_volume,
                     iwarp=True,
                     weight=nonlinear_weight_file,
                     iniwarp=[warp_file],
-                    inilev=levels_minpatches[n_iter - 1] + 1,
-                    maxlev=level_or_minpatch,
+                    inilev=nonlinear_levels[n_iter - 1] + 1,
+                    maxlev=maxlev,
                     out_file=out_file,
                     **verbosity_quietness_kwargs)
-                    
-            else:
-                out_qwarp = qwarp2(
-                    in_file=centered_head_file,
-                    base_file=nwarp_adjusted_mean,
-                    noneg=True,
-                    iwarp=True,
-                    weight=nonlinear_weight_file,
-                    iniwarp=[warp_file],
-                    inilev=nonlinear_levels[-1] + 1, # not ideal
-                    minpatch=level_or_minpatch,
-                    out_file=out_file)
+
+            warped_files.append(out_qwarp.outputs.warped_source)
+            warp_files.append(out_qwarp.outputs.source_warp)
+            previous_warp_files = warp_files
+             
+    for minpatch in enumerate(nonlinear_minimal_patches):
+        
+        if nonlinear_levels == []:
```

Comment:
this condition should be before the minpatch loop, otherwise previous_warp_files would be always the affine files